main.md

```diff
@@ -401,6 +435,91 @@ known type but containing unknown elements or elements of the wrong type, the AS
 
 # Token Request
 
+The `authorization_details` request parameter can be used to specify the authorization details a client wants the AS to assign to an access token. The AS checks whether the underlying grant (in case of grant types `authorization_code`, `refresh_token`, ...) or the client's policy (in case of grant type `client_credential`) allows the issuance of an access token with the requested authorization details. Otherwise, the AS refuses the request with error code `invalid_authorization_details` (similar to `invalid_scope`).
+
+## Comparing authorization details
+
+Many actions in the OAuth protocol allow the AS and RS to make security decisions based on whether or not the request
+is asking for "more" or "less" than a previous, existing request. For example, upon refreshing a token, the client can
+ask for a new access token with "fewer permissions" than had been previously authorized by the resource owner.
+Since the nature of an authorization details request is based solely on the API or APIs that it is describing, there is not
+a simple means of comparing any two arbitrary authorization details requests. 
+Authorization servers should not rely on simple object comparison in most cases, as the intersection of some elements
+within a request could have side effects in the access rights granted, depending on how the API
+has been designed and deployed. This is a similar effect to the scope values used with some APIs.
+
+However, when comparing a new request to an existing request, authorization servers can use the same 
+processing techniques as used in granting the request in the first place to determine if a resource
+owner needs to authorize the request. The details of this comparison are dependent on the definition
+of the `type` of authorization request and outside the scope of this specification, but common patterns
+can be applied.
+
+This shall be illustrated using our running example. The example authorization request in (#authz_request), if approved by the user, resulted in the issuance of an authorization code associated with the privileges to 
```

Comment:
It’s clearer now that those are examples. I think the amount of text and level of detail has become a bit too much for the editorial part of the spec and I might cause readers to focus too much on this topic. I suggest to move most of the text into implementation considerations.

Reply:
@jricher any thoughts?

Reply:
This section already is an implementation consideration. I think it makes sense to leave it inside the spec where it is. If we're going to talk about the topic of comparison at all, then we should make sure to do a thorough job of that topic.

Reply:
It is part of the token request section now.

Reply:
Right, and the next section "Interaction with the resource" parameter fills a similar role, I think -- and so does the "Relationship with the 'scope' parameter" and "Relationship with the 'resource' parameter", but slightly differently. Maybe all of these should be moved into the implementation considerations section together? That could make sense.

Reply:
You are right, all those sections contain a mixture of normative text relevant for the respective request and implementation considerations. I'm fine with moving it forward as is. We can sort out this mixture in a latter step.

Reply:
@tlodderstedt that works for me.